business/core/home/home.go

```diff
@@ -0,0 +1,203 @@
+package home
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/ardanlabs/service/business/core/event"
+	"github.com/ardanlabs/service/business/core/user"
+	"github.com/ardanlabs/service/business/data/order"
+	"github.com/ardanlabs/service/business/data/transaction"
+	"github.com/ardanlabs/service/foundation/logger"
+	"github.com/google/uuid"
+)
+
+// Set of error variables for CRUD operations.
+var (
+	ErrNotFound    = errors.New("home not found")
+	ErrInvalidUser = errors.New("user not valid")
+)
+
+// Storer interface declares the behaviour this package needs to persist and
+// retrieve data
+type Storer interface {
+	ExecuteUnderTransaction(tx transaction.Transaction) (Storer, error)
+	Create(ctx context.Context, nh Home) error
+	Update(ctx context.Context, hme Home) error
+	Delete(ctx context.Context, hme Home) error
+	Query(ctx context.Context, filter QueryFilter, orderBy order.By, pageNumber int, rowsPerPage int) ([]Home, error)
+	Count(ctx context.Context, filter QueryFilter) (int, error)
+	QueryByID(ctx context.Context, homeID uuid.UUID) (Home, error)
+	QueryByUserID(ctx context.Context, userID uuid.UUID) ([]Home, error)
```

Comment:
This minics what we did in products so we are good. Just want to say this for the future. Anytime we bring back a collection we need to ask ourselves if we need paging. I don't expect a user to have thousands of homes, but if they did we could be in trouble.

Reply:
I added paging in case we want to have an admin view of all homes. Let me know and I'll change it!

Reply:
The Query call should have paging. I was talking about QueryByUserID since it also returns a collection. Don't change anything at this time.